include/k4a/k4a.hpp

```diff
@@ -836,6 +842,14 @@ class transformation
      */
     void depth_image_to_color_camera(const image &depth_image, image *transformed_depth_image) const
     {
+        if (transformed_depth_image)
```

Comment:
We need to retain the ability to transform in to an existing output image. This enables callers to re-use the output image during a processing loop and not need to allocate each time. I think with this version, the image passed in by the caller will be lost and a new image will be allocated.

I think we should remove the creation of the new image in this overload, and only have it create the image in the new overload. This should make it more clear that in one version you are transforming to an existing image, and in the other you are creating a new one.

@williampMSFT, what do you think?

Reply:
I agree - it makes it much clearer that one is allocating memory on your behalf and the other isn't, which is relevant if you're trying to tune perf.

---
In reply to: [311126657](https://github.com/microsoft/Azure-Kinect-Sensor-SDK/pull/596#discussion_r311126657) [](ancestors = 311126657)

Reply:
I'll fix it. Thanks,